lagom-renderers/javadsl/src/main/scala/com/lightbend/lagom/spec/render/model/JavaEnumRender.scala

```diff
@@ -0,0 +1,60 @@
+package com.lightbend.lagom.spec.render.model
+
+import com.lightbend.lagom.spec.model.{ CustomEnum, Service }
+
+/**
+ *
+ */
+object JavaEnumRender {
+  val jacksonImports = "import com.fasterxml.jackson.annotation.*;"
+
+  def packageDeclaration(service: Service): String = s"package ${service.`package`};"
+
+  private def enumValues(customEnum: CustomEnum) = {
+    customEnum.values.map(v => s"""    ${v.toUpperCase.replaceAll("-", "_")}("$v")""").mkString(",\n")
+  }
+
+  private def enumDefinition(customEnum: CustomEnum) = {
+    s"""public enum ${customEnum.className} {
+       |${enumValues(customEnum)};
+       |
+      |    private final String value;
+       |
+      |    ${customEnum.className}(String value) {
+       |      this.value = value;
+       |    }
+       |
+      |    @com.fasterxml.jackson.annotation.JsonValue
+       |    public String getValue() {
+       |      return value;
+       |    }
+       |
+      |    @Override
+       |    public String toString() {
+       |      return String.valueOf(value);
```

Comment:
> I just realized that one difference is in the way it handles null, but I don't think value can be null here.

You are right.
I was trying to build an argument in favor of the current approach for supporting `UNKNOWN` and even then it won't make sense.